### ISSUE TITLE
Rehosting server

### ISSUE THREAD

**Issue by cpypcy** [mushroomhostage/exphc#104]
Well thing is i have my own dedicated server, a bit quadcore 12GB ram machine, only 100/100mbit connection, but it's pretty decent for 10-30 people (can handle quite load). I hosted few servers before.

Also what the hell man? No prior warning? Just shutting it down? Were you drinking?

**Comment by mushroomhostage** [comment]
Yeah, sudden shutdown I know, just didn't see any use in giving advance notice. But on the plus side, it means the final map is preserved as if no one knew it was shutting down (as opposed to the "end of the world parties" that some servers have in the final days of the map), which may make it easier for someone to continue it :)

For sure, you can host and run a successor to ExpHC if you'd like. All I ask is you use a different name for the server. The machine you have sounds like it should suffice; ExpHC was on a virtual dedicated server, 8-core CPU, 1 Gbps uplink, but only 8 GB RAM (6 GB dedicated to Minecraft). 

Will be in touch over PM to help set up. Note however, I'm planning on releasing the map publicly, to anyone interested in downloading it (not yet posted - its huge). So if you want to continue with the same map, just keep in mind anyone will be able to download it and see where everything is. Couple things that come to mind you could do - maybe give players a grace period to safely move to a new place on the map, or it may be easier to just start a new fresh map (which would actually be better than ExpHC's world, since most of the terrain is only Trees++ biomes, but if you regenerate there will also be awesome new ExtraBiomesXL biomes), possibly with some provision to allow players to move over their items or structures. Up to you - but I'll do what I can to help get it setup.

**Comment by cpypcy** [comment]
Ok fine, new server name, no problem, but what about not releasing map to public? I mean people worked hard on some bases and it would suck to start over again.
Anyway yeah i would like fresh start too, but i'm not sure about others, i was thinking to setup same server so people can  just continue like nothing happened, only there will be new hostname, that's all.

I'm running testing server on pashc.zapto.org to see if everything is working.

You did not have cake miners apocalypse plugin? Because i can't find fallout shelters.

**Comment by NewMilenium** [comment]
"I mean people worked hard on some bases and it would suck to start over again.
 Anyway yeah i would like fresh start too, but i'm not sure about others, i was thinking to setup same server so people can just continue like nothing happened, only there will be new hostname, that's all."

I totally support this opinion, and so i'm asking you too, mushroomhostage : please, do not release the map publicly.

**Comment by mushroomhostage** [comment]
World has been posted at https://github.com/mushroomhostage/exphc/wiki/Archive . I've always planned to release the final map to the public, for many reasons. Some players may wish to have a copy of their base (in fact one player actually installed a world-downloading mod for this purpose), including those that haven't been active lately. There have been 200+ whitelisted players, 5-10 active simultaneously; I would not want to deny the former players a copy of their portion of the world - especially if they haven't played for months and are curious if its still intact (I know multiple players were this is the case). Its become sort of a tradition in hardcore servers to release the map when the server ends – and ExpHC has ended. In fact it may be more enjoyable for some to roam the old world freely in single-player, exploring the whole map for interesting structures, possibly more enjoyable than playing the game for some. To see all the abandoned bases, active or destroyed bases, various buildings, whatever anyone felt compelled to build and whatever could survive in a dangerous post-apocalyptic environment. The ruins of the apocalypse, so to speak. I know I'm at least very interested to see what effect several months plus hundreds of players has had on a modded hardcore world. Releasing the map allows everyone to see this result if they are so inclined. 

The downside of course is preserving continuity in the transition to a new server, but I think the benefits of releasing the map outweigh this one downside. I realize this problem. Some suggestions though.. you could maybe host both maps, allow players to sign into the old map and collect say an inventory worth of items from the ExpHC map, then copy it to your map. Possibly use WorldEdit to bring over player structures, on request? Could be a lot of work I know. Sorry about that, but I really do want to publish the map for everyone's benefit.

And its not so bad having a new map.. look on the bright side :). For one it gives you an opportunity to address some of the balance issues with the ExpHC world. Fallout shelters having large numbers of solar arrays, extremely overpowered, for example. To answer your question the shelters on ExpHC were generated using [Cake's Miner Apocalypse](http://dev.bukkit.org/server-mods/cakes-miner-apocalypse/) in Minecraft 1.2.3 using the custom version I posted in the thread. The map was carried over to 1.2.5, along with the fallout shelters and craters. Unfortunately Cake's Miner Apocalypse has not been updated for 1.2.5 :(. Which is one reason why I removed the plugin itself, though kept the existing shelters. You may want to investigate other ways to generate structures. I placed the rare abandoned factories at random locations using WorldEdit, but of course that does not scale too well. I was planning to look into a better way to randomly generate structures (from schematics?) with random loot (in custom chests??) but did not get a chance to. But there must be a better way to do it, wish I had more time to investigate further, but I think you could come up with something.
